server/facets/resolveScopeIds.ts

```diff
@@ -0,0 +1,151 @@
+import * as types from 'types';
+import { Collection, CollectionPub, Pub } from 'server/models';
+import { bucketBy, indexById } from 'utils/arrays';
+import { getPrimaryCollection } from 'utils/collections/primary';
+import { assert } from 'utils/assert';
+
+import { ScopeKind, ByScopeKind, createByScopeKind } from '../../facets';
+
+export type ScopeStack = {
```

Comment:
super not important, I feel `ScopeStack` sounds like an actual stack is being made and that it should be treated as such. Is this true?